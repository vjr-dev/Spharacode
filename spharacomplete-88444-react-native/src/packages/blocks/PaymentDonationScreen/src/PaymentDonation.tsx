// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import React from "react";
import {
  Dimensions,
  Image,
  ImageBackground,
  KeyboardAvoidingView,
  Modal,
  Platform,
  SafeAreaView,
  StatusBar,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import { RFValue } from "react-native-responsive-fontsize";
import { heightPercentageToDP as hp } from "react-native-responsive-screen";
import { COLORS } from "../../../framework/src/Globals";
import {
  apple_pay,
  back1,
  back2,
  google_pay,
  image_back,
  next,
  tick,
} from "./assets";
import PaymentDonationController, { Props } from "./PaymentDonationController";
const windowWidth = Dimensions.get("window").width;
const windowHeight = Dimensions.get("window").height;

export default class PaymentDonation extends PaymentDonationController {
  constructor(props: Props) {
    super(props);
  }

  render() {
    return (
      <SafeAreaView style={styles.container1}>
        <View style={styles.child}>
          <ImageBackground source={back1} style={styles.image}>
            <ImageBackground source={back2} style={styles.image2}>
              <KeyboardAvoidingView
                keyboardVerticalOffset={1}
                behavior={"height"}
              >
                <StatusBar backgroundColor={"#f17234"} barStyle="default" />
                <View
                  style={{
                    height: hp("7%"),
                    width: windowWidth,
                    flexDirection: "row",
                    backgroundColor: COLORS.darkorange,
                    alignItems: "center",
                  }}
                >
                  <TouchableOpacity
                    style={{
                      width: "15%",
                      height: "90%",
                      justifyContent: "center",
                      alignItems: "center",
                    }}
                    testID="backButton"
                    onPress={() => this.goback()}
                  >
                    <Image
                      source={image_back}
                      style={{
                        height: 15,
                        width: 15,
                        tintColor: COLORS.backgroundGray,
                      }}
                    />
                  </TouchableOpacity>
                  <View
                    style={{
                      height: "100%",
                      width: "70%",
                      alignItems: "center",
                      justifyContent: "center",
                    }}
                  >
                    <Text
                      style={{
                        color: COLORS.backgroundGray,
                        fontWeight: "700",
                        fontSize: RFValue(16),
                      }}
                    >
                      Make payement of
                    </Text>
                  </View>
                  <View style={{ height: "100%", width: "15%" }} />
                </View>
                <View
                  style={{
                    height: 120,
                    width: windowWidth,
                    backgroundColor: COLORS.darkorange,
                    alignItems: "center",
                    justifyContent: "center",
                    alignSelf: "center",
                  }}
                >
                  <Text style={{ color: COLORS.white, fontSize: 35 }}>
                    â‚¹ {this.state.Amount}
                  </Text>
                  {!this.state.paymentdone ? (
                    <View
                      style={{
                        backgroundColor: COLORS.orangelight,
                        paddingHorizontal: 20,
                        borderRadius: 50,
                        alignItems: "center",
                        justifyContent: "center",
                        marginTop: 5,
                      }}
                    >
                      <Text style={{ fontSize: 15 }}>pending..</Text>
                    </View>
                  ) : null}
                </View>

                <View
                  style={{
                    height: 50,
                    width: windowWidth - 20,
                    alignSelf: "center",
                    alignItems: "center",
                    justifyContent: "center",
                    marginTop: 25,
                  }}
                >
                  <Text style={{ color: COLORS.ultralightwhite }}>
                    Send payment using any of the following methods.
                  </Text>
                </View>
                <TouchableOpacity
                  testID="gPayButton"
                  onPress={() => this.gpayclick()}
                  style={{
                    height: 60,
                    width: windowWidth - 20,
                    alignSelf: "center",
                    backgroundColor: COLORS.Viewback,
                    borderRadius: 50,
                    justifyContent: "space-between",
                    flexDirection: "row",
                    marginTop: 25,
                  }}
                >
                  <Image
                    source={google_pay}
                    style={{
                      backgroundColor: COLORS.white,
                      height: 40,
                      width: 40,
                      marginLeft: 10,
                      borderRadius: 50,
                      alignSelf: "center",
                    }}
                  />

                  <Text
                    style={{
                      alignSelf: "center",
                      color: COLORS.ultralightwhite,
                      width: "65%",
                    }}
                  >
                    Google Pay
                  </Text>
                  <Image
                    source={next}
                    style={{
                      height: 30,
                      width: 30,
                      tintColor: COLORS.ultralightwhite,
                      alignSelf: "center",
                      marginRight: 10,
                      opacity: 0.5,
                    }}
                  />
                </TouchableOpacity>
                <TouchableOpacity
                  testID="rPaybutton"
                  onPress={() => this.Rpay()}
                  style={{
                    height: 60,
                    width: windowWidth - 20,
                    alignSelf: "center",
                    backgroundColor: COLORS.Viewback,
                    borderRadius: 50,
                    justifyContent: "space-between",
                    flexDirection: "row",
                    marginTop: 25,
                  }}
                >
                  <Image
                    // source={google_pay}
                    style={{
                      backgroundColor: COLORS.white,
                      height: 40,
                      width: 40,
                      marginLeft: 10,
                      borderRadius: 50,
                      alignSelf: "center",
                    }}
                  />

                  <Text
                    style={{
                      alignSelf: "center",
                      color: COLORS.ultralightwhite,
                      width: "65%",
                    }}
                  >
                    REZORPAY
                  </Text>
                  <Image
                    source={next}
                    style={{
                      height: 30,
                      width: 30,
                      tintColor: COLORS.ultralightwhite,
                      alignSelf: "center",
                      marginRight: 10,
                      opacity: 0.5,
                    }}
                  />
                </TouchableOpacity>
                <TouchableOpacity
                  testID="stripePayButton"
                  onPress={() => this._checkSessionAPIcall()}
                  style={{
                    height: 60,
                    width: windowWidth - 20,
                    alignSelf: "center",
                    backgroundColor: COLORS.Viewback,
                    borderRadius: 50,
                    justifyContent: "space-between",
                    flexDirection: "row",
                    marginTop: 25,
                  }}
                >
                  <Image
                    // source={google_pay}
                    style={{
                      backgroundColor: COLORS.white,
                      height: 40,
                      width: 40,
                      marginLeft: 10,
                      borderRadius: 50,
                      alignSelf: "center",
                    }}
                  />

                  <Text
                    style={{
                      alignSelf: "center",
                      color: COLORS.ultralightwhite,
                      width: "65%",
                    }}
                  >
                    STRIPE PAY
                  </Text>
                  <Image
                    source={next}
                    style={{
                      height: 30,
                      width: 30,
                      tintColor: COLORS.ultralightwhite,
                      alignSelf: "center",
                      marginRight: 10,
                      opacity: 0.5,
                    }}
                  />
                </TouchableOpacity>
                {Platform.OS === "ios" ? (
                  <TouchableOpacity
                    testID="applePayButton"
                    onPress={() => this.applepayclick()}
                    style={{
                      height: 60,
                      width: windowWidth - 20,
                      alignSelf: "center",
                      backgroundColor: COLORS.Viewback,
                      borderRadius: 50,
                      justifyContent: "space-between",
                      flexDirection: "row",
                      marginTop: 25,
                    }}
                  >
                    <Image
                      source={apple_pay}
                      style={{
                        backgroundColor: COLORS.white,
                        height: 40,
                        width: 40,
                        marginLeft: 10,
                        borderRadius: 50,
                        alignSelf: "center",
                      }}
                    />

                    <Text
                      style={{
                        alignSelf: "center",
                        color: COLORS.ultralightwhite,
                        width: "65%",
                      }}
                    >
                      Apple pay
                    </Text>
                    <Image
                      source={next}
                      style={{
                        height: 30,
                        width: 30,
                        tintColor: COLORS.ultralightwhite,
                        alignSelf: "center",
                        marginRight: 10,
                        opacity: 0.5,
                      }}
                    />
                  </TouchableOpacity>
                ) : null}
                <Modal transparent={true} visible={this.state.Modal}>
                  <View
                    style={{
                      height: "100%",
                      width: "100%",
                      backgroundColor: COLORS.darkorange,
                      alignItems: "center",
                      justifyContent: "center",
                      alignSelf: "center",
                    }}
                  >
                    <Image
                      style={{ height: 100, width: 100, alignSelf: "center" }}
                      source={tick}
                    />
                    <Text style={{ fontSize: 35, marginTop: 75 }}>
                      Successful!
                    </Text>
                    <Text
                      style={{
                        fontSize: 18,
                        color: COLORS.white,
                        marginTop: 5,
                      }}
                    >
                      You donate â‚¹ {this.state.Amount} to {this.state.to}
                    </Text>
                    <TouchableOpacity
                      testID="doneButton"
                      onPress={() => this.successfulclick()}
                      style={{
                        height: 50,
                        width: windowWidth - 20,
                        alignItems: "center",
                        justifyContent: "center",
                        borderWidth: 1,
                        borderRadius: 50,
                        marginTop: 75,
                        borderStyle: "dashed",
                      }}
                    >
                      <Text>DONE</Text>
                    </TouchableOpacity>
                  </View>
                </Modal>
              </KeyboardAvoidingView>
            </ImageBackground>
          </ImageBackground>
        </View>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: windowHeight,
    width: windowWidth,
    backgroundColor: COLORS.white,
  },
  container1: { flex: 1, backgroundColor: COLORS.darkorange },
  child: { width: windowWidth },

  image: {
    height: windowHeight,
    width: windowWidth,
  },
  image2: {
    height: "100%",
    width: windowWidth,
    alignItems: "center",
  },
  btnview: {
    alignItems: "center",
    justifyContent: "center",
  },
  btn: {
    backgroundColor: "#FF0100",
    height: 60,
    width: 330,
    marginTop: 80,
    bottom: 30,
    alignItems: "center",
    justifyContent: "center",
    borderRadius: 30,
  },
  btntxt: {
    fontSize: 15,
    color: "#ffffff",
    letterSpacing: 0.5,
  },
});
// Customizable Area End
