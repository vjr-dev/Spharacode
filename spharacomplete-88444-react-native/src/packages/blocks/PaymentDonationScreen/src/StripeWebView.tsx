// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import React from "react";
import { StyleSheet, SafeAreaView, StatusBar } from "react-native";
import StripeWebViewController, { Props } from "./StripeWebViewController";
import { WebView } from "react-native-webview";

export default class StripeWebView extends StripeWebViewController {
  constructor(props: Props) {
    super(props);
  }

  render() {
    const url = this.props.route.params?.url?.url;
    return (
      <SafeAreaView style={styles.container}>
        <StatusBar
          barStyle="light-content"
          hidden={false}
          backgroundColor="#54595F"
          translucent={true}
        />
        <WebView
          source={{
            uri: url,
          }}
        />
      </SafeAreaView>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
// Customizable Area End
