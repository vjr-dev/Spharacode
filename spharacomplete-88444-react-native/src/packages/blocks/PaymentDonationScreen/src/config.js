// Customizable Area Start
Object.defineProperty(exports, "__esModule", {
    value: true,
});
exports.paymentDonationEndPoint = "bx_block_donationpayments/donationpayments";
// Customizable Area End