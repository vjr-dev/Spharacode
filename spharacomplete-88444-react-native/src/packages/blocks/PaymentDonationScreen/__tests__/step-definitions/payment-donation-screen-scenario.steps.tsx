//@ts-ignore
//@ts-nocheck
jest.useFakeTimers()
import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import PaymentDonation from "../../src/PaymentDonation";
import { Message } from "../../../../framework/src/Message";
import MessageEnum, { getName } from "../../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../../framework/src/RunEngine";
const screenProps = {
  navigation: {
    dispatch: jest.fn(),
    navigate: jest.fn(),
    goBack: jest.fn(),
    pop: jest.fn()
  },
  route:{
    params:{
      url:{
        url:"https://www.google.com/"
      }
    }
  },
  id: "PaymentDonationScreen"
};

const feature = loadFeature(
  "./__tests__/features/payment-donation-screen-scenario.feature"
);
defineFeature(feature, test => {
  beforeEach(() => {
    jest.resetModules();
    jest.mock("react-native/Libraries/Utilities/Platform", () => ({
      OS: "android",
      select: () => null,
    }));
 
    jest.mock('react-native-google-pay', () => ({
    
    }));
   
  });

  test("User navigates to Payment Donation Screen", ({ given, when, then }) => {
    let paymentDonationWrapper: ShallowWrapper;
    let instance :PaymentDonation;
    
    given("I am a User attempting to Payment Donation", () => {
      paymentDonationWrapper = shallow(<PaymentDonation {...screenProps} />);
      expect(paymentDonationWrapper).toBeTruthy();
      expect(paymentDonationWrapper).toMatchSnapshot();
    });

    when("I navigate to the Payment Donation Screen", () => {
      instance = paymentDonationWrapper.instance() as PaymentDonation;
      instance.setState({
        data: [1, 2, 3, 4, 5],
        Amount: 1,
        Modal: false,
        to: "Police officers",
        STRIPE_PUBLIC_KEY: "",
        CHECKOUT_SESSION_ID: "",
        UserName: "",
        Nickname: "",
        UserNumber: "",
        UserEmail: "",
        currentPosition: "",
        paymentdone: false,
      })
      instance.temp ={name:"abc"}
      expect(paymentDonationWrapper).toBeTruthy();
      expect(paymentDonationWrapper).toMatchSnapshot();

     
    });

    then("I can press back button", async () => {
      let backButton = paymentDonationWrapper.findWhere(
        node => node.prop("testID") === "backButton"
      );
      backButton.simulate("press");
      expect(backButton).toBeTruthy();

      
      
      
    });

    then("I can press google pay button", async () => {
      let gPayButton = paymentDonationWrapper.findWhere(
        node => node.prop("testID") === "gPayButton"
      );
      gPayButton.simulate("press");
      expect(gPayButton).toBeTruthy();
    });

    then("I can press Rpay button", async () => {
      let rPaybutton = paymentDonationWrapper.findWhere(
        node => node.prop("testID") === "rPaybutton"
      );
      rPaybutton.simulate("press");
      expect(rPaybutton).toBeTruthy();
   
    });

    then("I can press stripe pay button", async () => {
      let stripePayButton = paymentDonationWrapper.findWhere(
        node => node.prop("testID") === "stripePayButton"
      );
      stripePayButton.simulate("press");
      expect(stripePayButton).toBeTruthy();
    });

    then("I can press apple pay button", async () => {

      instance.applepayclick()
   
      let applePayButton = paymentDonationWrapper.findWhere(
        node => node.prop("testID") === "applePayButton" 
      );
      expect(applePayButton).toBeTruthy();
    });

    then("I can press done button", async () => {
      let doneButton = paymentDonationWrapper.findWhere(node => node.prop("testID") === "doneButton"
      );
      doneButton.simulate("press");
      expect(doneButton).toBeTruthy();
    });
    
    then("SessionAPI call with out errors", () => {
      let httpBody = JSON.stringify({
        data: {
          attributes: {
            donatelist_id: 1,
            amount: 100,
            success_url: "https://example.com/success",
            cancel_url: "https://example.com/cancel",
          },
        },
      });

      const sessionAPI: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      sessionAPI.addData(
        getName(MessageEnum.RestAPIResponceDataMessage),
        sessionAPI.messageId
      );
      sessionAPI.addData(
        getName(MessageEnum.RestAPIRequestBodyMessage),
        JSON.stringify(httpBody)
      );
      sessionAPI.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        JSON.parse(JSON.stringify({donation_payment:200,checkout_session:"abc.com"}))
      );
      instance.checkSessionAPIcallId = sessionAPI.messageId;
      runEngine.sendMessage("Unit Test", sessionAPI);
    });

  });
});
