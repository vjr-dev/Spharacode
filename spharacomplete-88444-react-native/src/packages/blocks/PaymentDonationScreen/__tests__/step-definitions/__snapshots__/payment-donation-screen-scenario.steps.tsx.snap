// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Donation Screen User navigates to Payment Donation Screen 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "rgba(255,138,73,1)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 750,
      }
    }
  >
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/image_background-1.png",
        }
      }
      style={
        Object {
          "height": 1334,
          "width": 750,
        }
      }
    >
      <ImageBackground
        source={
          Object {
            "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/image_background-2.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": "100%",
            "width": 750,
          }
        }
      >
        <KeyboardAvoidingView
          behavior="height"
          enabled={true}
          keyboardVerticalOffset={1}
        >
          <StatusBar
            animated={false}
            backgroundColor="#f17234"
            barStyle="default"
            showHideTransition="fade"
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(255,138,73,1)",
                "flexDirection": "row",
                "height": undefined,
                "width": 750,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "height": "90%",
                  "justifyContent": "center",
                  "width": "15%",
                }
              }
              testID="backButton"
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/image_back.png",
                  }
                }
                style={
                  Object {
                    "height": 15,
                    "tintColor": "#404040",
                    "width": 15,
                  }
                }
              />
            </TouchableOpacity>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "70%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#404040",
                    "fontSize": undefined,
                    "fontWeight": "700",
                  }
                }
              >
                Make payement of
              </Text>
            </View>
            <View
              style={
                Object {
                  "height": "100%",
                  "width": "15%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "rgba(255,138,73,1)",
                "height": 120,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontSize": 35,
                }
              }
            >
              â‚¹ 
              1
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255,211,154,1)",
                  "borderRadius": 50,
                  "justifyContent": "center",
                  "marginTop": 5,
                  "paddingHorizontal": 20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 15,
                  }
                }
              >
                pending..
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": 50,
                "justifyContent": "center",
                "marginTop": 25,
                "width": 730,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#a1a1a1",
                }
              }
            >
              Send payment using any of the following methods.
            </Text>
          </View>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#515151",
                "borderRadius": 50,
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-between",
                "marginTop": 25,
                "width": 730,
              }
            }
            testID="gPayButton"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/google_pay.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "rgb(255,255,255)",
                  "borderRadius": 50,
                  "height": 40,
                  "marginLeft": 10,
                  "width": 40,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#a1a1a1",
                  "width": "65%",
                }
              }
            >
              Google Pay
            </Text>
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/next.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 30,
                  "marginRight": 10,
                  "opacity": 0.5,
                  "tintColor": "#a1a1a1",
                  "width": 30,
                }
              }
            />
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#515151",
                "borderRadius": 50,
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-between",
                "marginTop": 25,
                "width": 730,
              }
            }
            testID="rPaybutton"
          >
            <Image
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "rgb(255,255,255)",
                  "borderRadius": 50,
                  "height": 40,
                  "marginLeft": 10,
                  "width": 40,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#a1a1a1",
                  "width": "65%",
                }
              }
            >
              REZORPAY
            </Text>
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/next.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 30,
                  "marginRight": 10,
                  "opacity": 0.5,
                  "tintColor": "#a1a1a1",
                  "width": 30,
                }
              }
            />
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#515151",
                "borderRadius": 50,
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-between",
                "marginTop": 25,
                "width": 730,
              }
            }
            testID="stripePayButton"
          >
            <Image
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "rgb(255,255,255)",
                  "borderRadius": 50,
                  "height": 40,
                  "marginLeft": 10,
                  "width": 40,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#a1a1a1",
                  "width": "65%",
                }
              }
            >
              STRIPE PAY
            </Text>
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/next.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 30,
                  "marginRight": 10,
                  "opacity": 0.5,
                  "tintColor": "#a1a1a1",
                  "width": 30,
                }
              }
            />
          </TouchableOpacity>
          <Component
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "rgba(255,138,73,1)",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/tick.png",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 100,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "fontSize": 35,
                    "marginTop": 75,
                  }
                }
              >
                Successful!
              </Text>
              <Text
                style={
                  Object {
                    "color": "rgb(255,255,255)",
                    "fontSize": 18,
                    "marginTop": 5,
                  }
                }
              >
                You donate â‚¹ 
                1
                 to 
                Police officers
              </Text>
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 50,
                    "borderStyle": "dashed",
                    "borderWidth": 1,
                    "height": 50,
                    "justifyContent": "center",
                    "marginTop": 75,
                    "width": 730,
                  }
                }
                testID="doneButton"
              >
                <Text>
                  DONE
                </Text>
              </TouchableOpacity>
            </View>
          </Component>
        </KeyboardAvoidingView>
      </ImageBackground>
    </ImageBackground>
  </View>
</SafeAreaView>
`;

exports[`Payment Donation Screen User navigates to Payment Donation Screen 2`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "rgba(255,138,73,1)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 750,
      }
    }
  >
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/image_background-1.png",
        }
      }
      style={
        Object {
          "height": 1334,
          "width": 750,
        }
      }
    >
      <ImageBackground
        source={
          Object {
            "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/image_background-2.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": "100%",
            "width": 750,
          }
        }
      >
        <KeyboardAvoidingView
          behavior="height"
          enabled={true}
          keyboardVerticalOffset={1}
        >
          <StatusBar
            animated={false}
            backgroundColor="#f17234"
            barStyle="default"
            showHideTransition="fade"
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(255,138,73,1)",
                "flexDirection": "row",
                "height": undefined,
                "width": 750,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "height": "90%",
                  "justifyContent": "center",
                  "width": "15%",
                }
              }
              testID="backButton"
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/image_back.png",
                  }
                }
                style={
                  Object {
                    "height": 15,
                    "tintColor": "#404040",
                    "width": 15,
                  }
                }
              />
            </TouchableOpacity>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "70%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#404040",
                    "fontSize": undefined,
                    "fontWeight": "700",
                  }
                }
              >
                Make payement of
              </Text>
            </View>
            <View
              style={
                Object {
                  "height": "100%",
                  "width": "15%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "rgba(255,138,73,1)",
                "height": 120,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontSize": 35,
                }
              }
            >
              â‚¹ 
              1
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255,211,154,1)",
                  "borderRadius": 50,
                  "justifyContent": "center",
                  "marginTop": 5,
                  "paddingHorizontal": 20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 15,
                  }
                }
              >
                pending..
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": 50,
                "justifyContent": "center",
                "marginTop": 25,
                "width": 730,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#a1a1a1",
                }
              }
            >
              Send payment using any of the following methods.
            </Text>
          </View>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#515151",
                "borderRadius": 50,
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-between",
                "marginTop": 25,
                "width": 730,
              }
            }
            testID="gPayButton"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/google_pay.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "rgb(255,255,255)",
                  "borderRadius": 50,
                  "height": 40,
                  "marginLeft": 10,
                  "width": 40,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#a1a1a1",
                  "width": "65%",
                }
              }
            >
              Google Pay
            </Text>
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/next.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 30,
                  "marginRight": 10,
                  "opacity": 0.5,
                  "tintColor": "#a1a1a1",
                  "width": 30,
                }
              }
            />
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#515151",
                "borderRadius": 50,
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-between",
                "marginTop": 25,
                "width": 730,
              }
            }
            testID="rPaybutton"
          >
            <Image
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "rgb(255,255,255)",
                  "borderRadius": 50,
                  "height": 40,
                  "marginLeft": 10,
                  "width": 40,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#a1a1a1",
                  "width": "65%",
                }
              }
            >
              REZORPAY
            </Text>
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/next.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 30,
                  "marginRight": 10,
                  "opacity": 0.5,
                  "tintColor": "#a1a1a1",
                  "width": 30,
                }
              }
            />
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#515151",
                "borderRadius": 50,
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-between",
                "marginTop": 25,
                "width": 730,
              }
            }
            testID="stripePayButton"
          >
            <Image
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "rgb(255,255,255)",
                  "borderRadius": 50,
                  "height": 40,
                  "marginLeft": 10,
                  "width": 40,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#a1a1a1",
                  "width": "65%",
                }
              }
            >
              STRIPE PAY
            </Text>
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/next.png",
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 30,
                  "marginRight": 10,
                  "opacity": 0.5,
                  "tintColor": "#a1a1a1",
                  "width": 30,
                }
              }
            />
          </TouchableOpacity>
          <Component
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "rgba(255,138,73,1)",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../packages/blocks/PaymentDonationScreen/assets/tick.png",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 100,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "fontSize": 35,
                    "marginTop": 75,
                  }
                }
              >
                Successful!
              </Text>
              <Text
                style={
                  Object {
                    "color": "rgb(255,255,255)",
                    "fontSize": 18,
                    "marginTop": 5,
                  }
                }
              >
                You donate â‚¹ 
                1
                 to 
                Police officers
              </Text>
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 50,
                    "borderStyle": "dashed",
                    "borderWidth": 1,
                    "height": 50,
                    "justifyContent": "center",
                    "marginTop": 75,
                    "width": 730,
                  }
                }
                testID="doneButton"
              >
                <Text>
                  DONE
                </Text>
              </TouchableOpacity>
            </View>
          </Component>
        </KeyboardAvoidingView>
      </ImageBackground>
    </ImageBackground>
  </View>
</SafeAreaView>
`;
